Azure/sonic-utilities
# [db_migrator][201911] Support shared headroom in db_migrator on Mellanox platform

Added support to upgrade a switch from old version to one with shared headroom pool support

**- What I did**
- Generic update: advance the version of database to 1.5.0
- Mellanox specific change:
  - support shared headroom pool for MSFT SKUs
  - extend the infrastructure of Mellanox buffer migratory:
    - support mapping from old buffer pool configuration set to new one by SKU
    - support migrate headroom info based on SKU
    - condemns the representation of buffer pool configuration to reduce redundant code
    - organize the configuration dict in a better way

**- How I did it**

**- How to verify it**
Verify upgrade from 201811 to 201911 with shared headroom functionality
Verify upgrade from early 201911 to 201911 with shared headroom functionality

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
